// plugin-welcome/src/api/models/EntPatientofphysician.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPatientofphysicianEdges,
    EntPatientofphysicianEdgesFromJSON,
    EntPatientofphysicianEdgesFromJSONTyped,
    EntPatientofphysicianEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPatientofphysician
 */
export interface EntPatientofphysician {
    /**
     * Ailment holds the value of the "Ailment" field.
     * @type {string}
     * @memberof EntPatientofphysician
     */
    ailment?: string;
    /**
     * 
     * @type {EntPatientofphysicianEdges}
     * @memberof EntPatientofphysician
     */
    edges?: EntPatientofphysicianEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPatientofphysician
     */
    id?: number;
}

export function EntPatientofphysicianFromJSON(json: any): EntPatientofphysician {
    return EntPatientofphysicianFromJSONTyped(json, false);
}

export function EntPatientofphysicianFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPatientofphysician {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'ailment': !exists(json, 'Ailment') ? undefined : json['Ailment'],
        'edges': !exists(json, 'edges') ? undefined : EntPatientofphysicianEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntPatientofphysicianToJSON(value?: EntPatientofphysician | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'Ailment': value.ailment,
        'edges': EntPatientofphysicianEdgesToJSON(value.edges),
        'id': value.id,
    };
}
